File: Samples/CSSample/Procotol/DOSSampleSpaceShipStructs.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.InteropServices;
using System.Xml;
using System.Text;
using Newtonsoft.Json.Linq;
#if ENABLE_XLS_PROCESS
using System.Windows.Forms;
using Spire.Xls;
using DataConvertor;
#endif
using DOSSystem;
using DOSSample;


namespace DOSSample
{
	//
	public class POS
	{
		public int		x;
		public int		y;
		
		
		public enum POS_MEMBER_IDS
		{
			SST_TEST_STRUCT_X = 1,
			SST_TEST_STRUCT_Y = 2,
			
		};
		
		public class POS_MEMBER_IDS_UTILS
		{
			public const int ENUM_COUNT = 2;
		};
		
		public POS()
		{
			Clear();
		}
		public void Clear()
		{
			x=0;
			y=0;
			
		}
		public bool MakePacket(CSmartStruct Packet)
		{
			uint FailCount=0;
	
			{
				if(!Packet.AddMember((ushort)POS_MEMBER_IDS.SST_TEST_STRUCT_X,x)) FailCount++;
			}
			
			{
				if(!Packet.AddMember((ushort)POS_MEMBER_IDS.SST_TEST_STRUCT_Y,y)) FailCount++;
			}
			
			
	
			return FailCount==0;
		}
		public bool ParsePacket(CSmartStruct Packet)
		{
			
			int Pos=Packet.GetFirstMemberPosition();
			while(Pos>0)
			{
				ushort MemberID=0;
				CSmartValue PacketMember=Packet.GetNextMember(ref Pos,ref MemberID);
				switch(MemberID)
				{
				case (ushort)POS_MEMBER_IDS.SST_TEST_STRUCT_X:
					{
						x=PacketMember;
					}
					break;
				case (ushort)POS_MEMBER_IDS.SST_TEST_STRUCT_Y:
					{
						y=PacketMember;
					}
					break;
				
				}
			}
			
			return true;
		}
		public uint GetSmartStructSize()
		{
			uint _PackSize = 0;
			_PackSize += CSmartStruct.GetFixMemberSize(sizeof(int));
			
			_PackSize += CSmartStruct.GetFixMemberSize(sizeof(int));
			
			
			return _PackSize;
		}
		public void CloneFrom(POS Value)
		{
			{
				x=Value.x;
			}
			{
				y=Value.y;
			}
			
		}
		public POS Clone()
		{
			POS CloneObject=new POS();
			CloneObject.CloneFrom(this);
			return CloneObject;
		}
		
		
		
	};
	
	
	
	
};

File: Samples/JSSample/src/Protocol/DOSSampleSpaceShipStructs.ts
import { CSmartValue, CSmartArray, CSmartStruct } from "../DOSSystem/SmartStruct";
import { OBJECT_ID } from "../DOSSystem/DOSSystem";
import { ALL_CLASS } from "./DOSSampleDataStructs";
import { SHIP_TYPE } from "./DOSSampleSpaceShipEnums"


//


export const enum POS_MEMBER_IDS
{
	SST_TEST_STRUCT_X = 1,
	SST_TEST_STRUCT_Y = 2,
	
};

export class POS_MEMBER_IDS_UTILS
{
	public static ENUM_COUNT: number = 2;
};

export class POS
{
	public x: number;
	public y: number;
	
	
	constructor()
	{
		this.x=0;
		this.y=0;
		
	}
	public Clear(): void
	{
		this.x=0;
		this.y=0;
		
	}
	public MakePacket(Packet: CSmartStruct): boolean
	{
		let FailCount=0;

		{
			if(!Packet.AddMemberInt(POS_MEMBER_IDS.SST_TEST_STRUCT_X,this.x)) FailCount++;
		}
		{
			if(!Packet.AddMemberInt(POS_MEMBER_IDS.SST_TEST_STRUCT_Y,this.y)) FailCount++;
		}
		

		return FailCount==0;
	}
	public ParsePacket(Packet: CSmartStruct): boolean
	{
		
		let Pos=Packet.GetFirstMemberPosition();
		while(Pos.Pos>0)
		{
			let PacketMember=Packet.GetNextMember(Pos);
			switch(Pos.MemberID)
			{
			case POS_MEMBER_IDS.SST_TEST_STRUCT_X:
				{
					this.x=PacketMember.valueOf();
				}
				break;
			case POS_MEMBER_IDS.SST_TEST_STRUCT_Y:
				{
					this.y=PacketMember.valueOf();
				}
				break;
			
			}
		}
		
		return true;
	}
	public GetSmartStructSize(): number
	{
		let _PackSize = 0;
		_PackSize += CSmartStruct.GetFixMemberSize(4);
		
		_PackSize += CSmartStruct.GetFixMemberSize(4);
		
		
		return _PackSize;
	}
	public CloneFrom(Value: POS): void
	{
		{
			if(Value.x)
			{
				this.x=Value.x;
			}
		}
		{
			if(Value.y)
			{
				this.y=Value.y;
			}
		}
		
	}
	public Clone(): POS
	{
		let CloneObject=new POS();
		CloneObject.CloneFrom(this);
		return CloneObject;
	}
};

ALL_CLASS["POS"] = POS;


